code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1632,3 +1668,30 @@ pub(crate) enum RewardAccumulationCalculationError {
 	/// to release for the rewards accumulated.
 	RewardsPotEmpty,
 }
+
+pub(crate) fn claim_of_stake<T: Config>(
+	stake: &StakeOf<T>,
+	total_shares: &T::Balance,
+	reward: &Reward<T::Balance>,
+	reward_asset_id: &<T as Config>::AssetId,
+) -> Result<T::Balance, DispatchError> {
+	let claim = if total_shares.is_zero() {
+		T::Balance::zero()
+	} else {
+		let inflation = stake.reductions.get(reward_asset_id).cloned().unwrap_or_else(Zero::zero);
+
+		// REVIEW(benluelo): Review expected rounding behaviour, possibly switching to the following
+		// implementation (or something similar):
+		// Perbill::from_rational(stake.share, *total_shares)
+		// 	.mul_floor(reward.total_rewards)
+		// 	.safe_sub(&inflation)?;
```

Comment:
+1 to the switch

Reply:
nice, ticket made: https://app.clickup.com/t/33e5a88

we can review this later this week :+1: